lib/KTable/KTableGridItem.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+
+  <td
+    :class="$computedClass(coreOutlineFocus)"
+    :style="{ textAlign: textAlign, minWidth: minWidth, width: width }"
+    tabindex="0"
+    role="gridcell"
+    @keydown="onKeydown"
+  >
+    <slot :content="content">
+      {{ content }}
+    </slot>
+  </td>
+
+</template>
+
+
+<script>
+
+  import { DATA_TYPE_NUMERIC } from '../composables/useSorting';
+
+  export default {
+    name: 'KTableGridItem',
+    props: {
+      content: {
+        type: [String, Number],
+        required: true,
+      },
+      dataType: {
+        type: String,
+        default: 'string',
+      },
+      rowIndex: {
+        type: Number,
+        required: true,
+      },
+      colIndex: {
+        type: Number,
+        required: true,
+      },
+      minWidth: {
+        type: String,
+        default: 'auto',
+      },
+      width: {
+        type: String,
+        default: 'auto',
+      },
+    },
+    computed: {
+      coreOutlineFocus() {
+        return {
+          ':focus': {
+            ...this.$coreOutline,
+            outlineOffset: '-2px',
+          },
+        };
+      },
+      textAlign() {
+        return this.dataType === DATA_TYPE_NUMERIC ? 'right' : 'left';
+      },
+    },
+    methods: {
+      onKeydown(event) {
+        // Check if the focused element is a button or actionable item
+        const focusedElement = event.target;
+        if (
+          event.key === 'Enter' &&
+          (focusedElement.tagName === 'BUTTON' ||
+            focusedElement.tagName === 'A' ||
+            focusedElement.hasAttribute('role'))
```

Comment:
Ah I see. As always you think broad and ahead :)! That's a good consideration. I think for such cases we'd need to call `focus` rather than `click` though? I don't quite see how `click` would help with non-clickable elements.

Reply:
With this I am not suggesting we do some branching here to call `focus`. I am not entirely sure how it should work for progress bars. It's just that `click` seems to make sense for clickable elements to me. Perhaps better remove for now and clarify later when the need arises? I think as soon as we start introducing the table in more places, we will always want to do a11y QA run on the feature anyways so it should surface. Generally KDS can't do everything on its own, particularly for content provided via slots - so this is quite a common approach.

Reply:
Okay not even `focus` needed per @radinamatic's recommendation